### PR TITLE
Add europe experiment

### PR DESCRIPTION
**Pull request by rowannekabalan**
## What does this change?

- This adds a new 0% experiment group for the Europe network front 
 
_And_

- Restricts access to the `/europe` network front page to participants in the experiment by intercepting the request in FaciaController and redirecting to a 404 page appropriately

 If you would like more background information on why we are doing this work you can check [here](https://docs.google.com/document/d/1654fwLrllHDIT6hEr_05Kv5W8Dh90pUhofHHPaAMS9w/edit).

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots

Before:
image

After:
image

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x ] Locally
- [ x] On CODE (optional)

### CODE REVIEW

File: common/app/experiments/Experiments.scala [hunk 1]
```diff
@@ -75,3 +75,12 @@ object DCROnwardsData
       sellByDate = LocalDate.of(2023, 6, 2),
       participationGroup = Perc0C,
     )
+
+object EuropeNetworkFront
+    extends Experiment(
+      name = "europe-network-front",
+      description = "Test new europe network front",
+      owners = Seq(Owner.withGithub("rowannekabalan")),
+      sellByDate = LocalDate.of(2022, 9, 30),
```
Comment (oliverlloyd) [inline common/app/experiments/Experiments.scala:84]: You might want to set this to a later date. Last I checked, expiring tests break the build and can be pesky

File: facia/app/controllers/FaciaController.scala [hunk 2]
```diff
@@ -175,7 +176,11 @@ trait FaciaController
     } else result
 
   import PressedPage.pressedPageFormat
-  private[controllers] def renderFrontPressResult(path: String)(implicit request: RequestHeader) = {
+  private[controllers] def renderFrontPressResult(path: String)(implicit request: RequestHeader): Future[Result] = {
```
Comment (oliverlloyd) [inline facia/app/controllers/FaciaController.scala:179]: I don't understand Scala very well so I just wanted to check what impact `: Future[Result]` has here, is this just a type check?

File: facia/app/controllers/FaciaController.scala [hunk 3]
```diff
@@ -175,7 +176,11 @@ trait FaciaController
...
+    val participatingInTest = ActiveExperiments.isParticipating(EuropeNetworkFront)
+    if (!participatingInTest && path == "europe") {
```
Comment (oliverlloyd) [inline facia/app/controllers/FaciaController.scala:181]:
```suggestion
    if (path == "europe" && !participatingInTest) {
```

Just a suggestion but my brains find this order easier to read

**Review by oliverlloyd — APPROVED** [review]
Just a few non blocking comment but otherwise looks great

File: common/app/experiments/Experiments.scala (hunk 1 above)
Comment (rowannekabalan) [reply common/app/experiments/Experiments.scala:84]: Thanks for the heads up! Will push it back another month

File: facia/app/controllers/FaciaController.scala (hunk 2 above)
Comment (rowannekabalan) [reply facia/app/controllers/FaciaController.scala:179]: Yes this is the function's expected return type. It is necessary as we are explicitly using a `return` statement which means this won't compile without an explicit return type

Comment (oliverlloyd) [reply facia/app/controllers/FaciaController.scala:179]: Okay, thanks for clarifying 👍

**Comment by oliverlloyd** [comment]
All looks good, I think you just need to run scala fmt to fix the TeamCity build

**Pull request merged**